src/components/organism/SwapTokens/index.tsx

```diff
@@ -57,42 +91,243 @@ const SwapTokens = () => {
+
+          setTokenAValueForSwap({ tokenValue: assetTokenWithSlippage });
+          setTokenBValueForSwap({ tokenValue: value });
+          setSelectedTokenAValue({ tokenValue: assetTokenNoDecimals });
+        }
+      }
     }
-    if ((selectedTokens.tokenA && selectedTokenAValue?.tokenValue <= 0) || selectedTokenBValue?.tokenValue <= 0) {
-      return "Enter Amount";
+  };
+
+  const getPriceOfAssetTokenBFromAssetTokenA = async (value: number) => {
+    if (api) {
+      const valueWithDecimals = formatInputTokenValue(value, selectedTokens.tokenA.decimals);
+      if (selectedTokens.tokenA.tokenId && selectedTokens.tokenB.tokenId) {
+        const assetTokenPrice = await getAssetTokenBFromAssetTokenA(
+          api,
+          valueWithDecimals,
+          selectedTokens.tokenA.tokenId,
+          selectedTokens.tokenB.tokenId
+        );
+
+        if (assetTokenPrice) {
+          const assetTokenNoSemicolons = assetTokenPrice.toString().replace(/[, ]/g, "");
+          const assetTokenNoDecimals = formatDecimalsFromToken(
+            parseFloat(assetTokenNoSemicolons),
+            selectedTokens.tokenB.decimals
+          );
+
+          const assetTokenWithSlippage = assetTokenNoDecimals - (assetTokenNoDecimals * slippageValue) / 100;
+
+          setTokenAValueForSwap({ tokenValue: value });
+          setTokenBValueForSwap({ tokenValue: assetTokenWithSlippage });
+          setSelectedTokenBValue({ tokenValue: assetTokenNoDecimals });
+        }
+      }
     }
-    if (
-      selectedTokens.tokenA &&
-      selectedTokens.tokenB &&
-      selectedTokenAValue?.tokenValue > selectedTokenBValue?.tokenValue
-    ) {
-      return "Insufficient DAI amount";
+  };
+
+  const tokenAValue = async (value: number) => {
+    const baseString = value.toString();
+    console.log("decimals:", selectedTokens.tokenA.decimals);
+    if (baseString.includes(".")) {
+      if (baseString.split(".")[1].length > parseInt(selectedTokens.tokenA.decimals)) {
+        console.log("too many decimals");
+        // todo: write error message
+        return;
+      }
     }
-    if (
-      selectedTokens.tokenA &&
-      selectedTokens.tokenB &&
-      selectedTokenAValue?.tokenValue <= selectedTokenBValue?.tokenValue
-    ) {
-      return "Swap";
+    setSelectedTokenAValue({ tokenValue: value });
+    setInputEdited({ inputType: "exactIn" });
+    if (selectedTokenAValue) {
+      if (selectedTokens.tokenA.tokenSymbol === TokenSelection.NativeToken) {
+        getPriceOfAssetTokenFromNativeToken(value);
+        if (tokenBalances?.balance) {
+          setWalletHasEnoughWnd(value <= tokenBalances?.balance ? true : false);
+        }
+      } else if (selectedTokens.tokenB.tokenSymbol === TokenSelection.NativeToken) {
+        getPriceOfNativeTokenFromAssetToken(value);
+        if (tokenBalances?.balance) {
+          setWalletHasEnoughWnd(value <= tokenBalances?.balance ? true : false);
```

Comment:
You can simplify this condition:
instead `value <= tokenBalances?.balance ? true : false` you can write `value <= tokenBalances?.balance`.